app/src/main/java/org/zalando/nakadi/service/job/DlqRedriveEventTypeAttachmentJob.java

```diff
@@ -0,0 +1,161 @@
+    private final SubscriptionClientFactory subscriptionClientFactory;
+    private final CursorConverter cursorConverter;
+    private final String dlqRedriveEventTypeName;
+    private final ExclusiveJobWrapper jobWrapper;
+
+    @Autowired
+    public DlqRedriveEventTypeAttachmentJob(
+            final FeatureToggleService featureToggleService,
+            final EventTypeCache eventTypeCache,
+            final TimelineService timelineService,
+            final SubscriptionDbRepository subscriptionRepository,
+            final SubscriptionClientFactory subscriptionClientFactory,
+            final CursorConverter cursorConverter,
+            @Value("${nakadi.dlq.redriveEventTypeName}") final String dlqRedriveEventTypeName,
+            @Value("${nakadi.jobs.dlqRedriveEventTypeAttach.runPeriodMs}") final int periodMs,
+            final JobWrapperFactory jobWrapperFactory) {
+        this.featureToggleService = featureToggleService;
+        this.eventTypeCache = eventTypeCache;
+        this.timelineService = timelineService;
+        this.subscriptionRepository = subscriptionRepository;
+        this.subscriptionClientFactory = subscriptionClientFactory;
+        this.cursorConverter = cursorConverter;
+        this.dlqRedriveEventTypeName = dlqRedriveEventTypeName;
+        this.jobWrapper = jobWrapperFactory.createExclusiveJobWrapper(JOB_NAME, periodMs);
+    }
+
+    @Scheduled(
+            fixedDelayString = "${nakadi.jobs.checkRunMs}",
+            initialDelayString = "${random.int(${nakadi.jobs.checkRunMs})}")
+    public void attachDlqRedriveEventType() {
+        if (featureToggleService.isFeatureEnabled(Feature.DLQ_REDRIVE_EVENT_TYPE_ATTACHMENT_JOB)) {
+            LOG.warn("Feature toggle is disabled for DLQ redrive event type attachment job: skipping the run.");
+            return;
+        }
+        try {
+            jobWrapper.runJobLocked(this::attachDlqRedriveEventTypeLocked);
+        } catch (final Exception e) {
+            LOG.error("Failed to run the job to attach DLQ redrive event type to corresponding subscriptions", e);
+        }
+    }
+
+    private void attachDlqRedriveEventTypeLocked() {
+
+        final List<String> orderedPartitions = eventTypeCache.getOrderedPartitions(dlqRedriveEventTypeName);
+
+        final List<NakadiCursor> beginCursors = orderedPartitions.stream()
+                .map(p -> new Cursor(p, Cursor.BEFORE_OLDEST_OFFSET))
+                .map(c -> cursorConverter.convert(dlqRedriveEventTypeName, c))
+                .collect(Collectors.toList());
+
+        // it's inefficient to re-consume from begin, but we expect this event type to be empty most of the time anyway
+        final HighLevelConsumer consumer = timelineService.createEventConsumer(JOB_NAME + "-job", beginCursors);
+
+        final Set<String> subscriptionIds = new HashSet<>();
+        while (true) {
+            final List<ConsumedEvent> events = consumer.readEvents();
```

Comment:
it will read from begin over and over again because it does not commit offsets. basically, this loop will run forever. it looks like you need to track offsets. I think it is better to reconsider the job approach.

Reply:
> it will read from begin over and over again because it does not commit offsets.

That's the point here, but most of the time the topic is completely empty, so why bother?

> basically, this loop will run forever. it looks like you need to track offsets. I think it is better to reconsider the job approach.

It will not run forever, it will only consume from begin till end on every job run.

Reply:
ah, actually, I am wrong, it should work just fine because consumer will manage it.